lib/arjdbc/jdbc/adapter.rb

```diff
@@ -289,12 +289,14 @@ def active?
 
       # @override
       def reconnect!
+        super
```

Comment:
It looks like ActiveRecord 4.2 has roughly the same code, so I would guess that the same issue exists in the stable release. My current codebase isn't in a state that I can easily test the stable branch, but I can dig into it if you think its something that should be fixed.